### DISCUSSION TITLE
Feature Proposal: Adding Buttons for Cell Data Decoding

### DISCUSSION THREAD

**Discussion by Flouse** [Magickbase/ckb-explorer-public-issues#324] [Ideas]
Recently, I created [a simple tutorial showing how to inscribe "Hello Common Knowledge Base!"](https://ckb-tthw.vercel.app/) or any message to CKB Testnet. @quake suggested a new feature for CKB Explorer. And I think it is worth discussing here.

## Problem
The data stored in CKB cells usually is not human-readable formats. It can be difficult for users to interpret this data.
![image](https://github.com/Magickbase/ckb-explorer-public-issues/assets/1297478/9773954c-7791-4c5f-9b29-688f0fc04d5f)
The [above cell data](https://pudge.explorer.nervos.org/transaction/0x39d6d7b6129b7e418c9ea6a353a5d85eb69f9ee5b4c7c43223fe0fad2b0e6200) is the hexadecimal format of "Hello Common Knowledge Base!".

#### Related issue
- https://github.com/Flouse/ckb-tthw/issues/1

## Proposed Feature 
We propose some addition of buttons for decoding cell data in the CKB Explorer, similar to Etherscan
![image](https://github.com/Magickbase/ckb-explorer-public-issues/assets/1297478/f86ae1d8-8584-496c-bb62-96194d7dfa4e)

These buttons would allow users to easily decode cell data to decimal or UTF-8 string formats.
This feature would be particularly useful for some on-chain scripts such as SUDT and DAO, as their cell data could be easily decoded and interpreted by users.

### Functionality
When a user clicks one of the decoding buttons, the corresponding decoder function would be applied to the cell data and the result would be displayed in a new modal or overlay. The decoded data could be displayed in plain text to make it easily accessible to users.

**Comment by homura** [comment]
Looks like a cool feature, I personally need to encode or decode some data, and before this feature is discussed, I recommend to use [CyberChef](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')&input=NDg2NTZjNmM2ZjIwNDM2ZjZkNmQ2ZjZlMjA0YjZlNmY3NzZjNjU2NDY3NjUyMDQyNjE3MzY1MjE)

<img width="1363" alt="image" src="https://github.com/Magickbase/ckb-explorer-public-issues/assets/7511174/6e7b5418-3931-483d-9444-97faf6c1a4e5">

**Comment by Keith-CY** [comment]
This feature is tracked by https://github.com/Magickbase/ckb-explorer-public-issues/issues/331
